src/test/java/calculator/ParserTest.java

```diff
@@ -0,0 +1,48 @@
+package calculator;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+
+class ParserTest {
+	Parser parser = new Parser();
+
+	@DisplayName("String을 Integer로 올바르게 파싱하는 지 점검")
+	@Test
+	void stringToInteger() {
+		String stringToInt = "3";
+		assertThat(parser.stringToInteger(stringToInt)).isEqualTo(3);
+	}
+
+	@DisplayName("인자가 문자일 때, 터지는 exception 점검.")
+	@Test
+	void shouldThrowException() {
+		String invalidValue = "j";
+		assertThat(parser.stringToInteger(invalidValue)).isEqualTo(new NumberFormatException());
+	}
+
+	@DisplayName("양수인지 점검")
+	@Test
+	void shouldCheckPositiveValueTest() {
+		int negativeInt = -1;
+		Parser mockedParser = mock(Parser.class);
+		doThrow(new IllegalArgumentException()).when(mockedParser).checkTheValueIsPositive(negativeInt);
```

Comment:
mock을 사용하지 않고 exception 테스트는 다음과 같이 고칠 수 있을 것 같아요!

```suggestion
		assertThatThrownBy(() -> new Parser().checkTheValueIsPositive(negativeInt))
		            .isInstanceOf(IllegalArgumentException.class);
```

Reply:
아하 넵 감사합니다 !

Reply:
그런데 왜 mock을 사용하는 것보다 assertThat사용을 더 권장하시는 지 궁금합니다!

Reply:
음..!
제가 리뷰를 이상하게 달아놨네여ㅎㅎ
mock을 사용하지마세요! 라는건 아니지만
`mockito` 의 `doThrow` 구문은 stub을 하는 역할입니다. 
어떤 말이냐면 `mock(Parser.class)`로 생성한 `Parser mockedParser`가 어떠한 행동을 할지 강제 할 수 있도록 하는 거에요!
그래서 지금 코드에서는 `mockedPaser`가 `-1`을 입력받았을 경우 `IlleagalArgumentException`을 발생하게 해라! 라고 지정을 한 것입니다.
이러한 mock 라이브러리를 이용한다면 아래와 같은 코드도 통과를 한답니다.

```java
	@DisplayName("양수인지 점검")
	@Test
	void shouldCheckPositiveValueTest() {
		int negativeInt = -1;
		Parser mockedParser = mock(Parser.class);
		doThrow(new IllegalArgumentException()).when(mockedParser).checkTheValueIsPositive(1);

		assertThatThrownBy(() -> mockedParser.checkTheValueIsPositive(1))
				.isInstanceOf(IllegalArgumentException.class);
	}
```

그래서 사실상 테스트가 의미있는 테스트가 아니게 되어버렸어요!

이전에 했던 리뷰를 수정하고 정리하자면.
`mockito`의 `doThrow` 는 단언문이 아닙니다. 그래서 위 테스트 코드에는 exception을 검증 하는 코드가 누락되어있습니다!

Reply:
아하 넵 감사합니다!!!